src/modules-load/modules-load.c

```diff
@@ -163,34 +177,43 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 log_warning_errno(r, "Failed to parse kernel command line, ignoring: %m");
 
-        r = module_setup_context(&ctx);
-        if (r < 0)
-                return log_error_errno(r, "Failed to initialize libkmod context: %m");
+        _cleanup_close_pair_ int seq[2] = EBADF_PAIR;
+        long ncpus = sysconf(_SC_NPROCESSORS_ONLN);
+        if (ncpus < 0)
+                log_warning_errno(r, "Failed to get number of cpus");
 
-        r = 0;
+        long max_tasks = MAX(MIN(ncpus / 2, 10), 2);
+        _cleanup_free_ pthread_t *threads = malloc(sizeof(pthread_t) * max_tasks);
```

Comment:
also, please use `new(pthread_t, max_tasks)`, i.e. use a typesafe and overflow-safe way to allocate